### PR TITLE
Update react-hook-form to 7.6.1

### LINKED ISSUE #2253: Forms: Upgrade React Hook Form to v7
Under the hood, Redwood's @redwoodjs/forms uses v6.15.4 of React Hook Forms, which is now at v7:
>React Hook Form focus on the following aspect on Version 7:
> - (DX) Strict typed form
> - Reduce package size
> - Performance enhancements
> - improve API's simplicity and consistency

Releases: https://github.com/react-hook-form/react-hook-form/releases

Upgrade Guide: https://react-hook-form.com/migrate-v6-to-v7/

### PR DESCRIPTION
**Pull request by morganmspencer**
Started updating react-hook-form to 7.0.7 to close #2253 

_Disclaimer: first contribution besides minor typo fixes_ 😂

I believe all necessary files are updated but I'm running into a configuration error with the tests. It appears to be occurring in the jest.config.js file, but I may have done something wrong.

Happy to make any other edits if I missed anything!

### CODE REVIEW

**Comment by sriverag** [comment]
@morganmspencer - This looks awesome! I suggest including the `yarn.lock` file because the Cypress E2E tests check is installing the dependencies using `yarn install --frozen-lockfile` probably causing the tests to fail.

**Comment by thedavidprice** [comment]
@morganmspencer could you update to the most recent React Hook Form version?
https://github.com/react-hook-form/react-hook-form/releases

pinging @cannikin for review

File: packages/forms/src/index.tsx [hunk 1]
```diff
@@ -268,7 +273,7 @@ const TextAreaField = forwardRef<
     <textarea
       {...tagProps}
       id={props.id || props.name}
-      ref={(element) => {
+      {...(element: HTMLTextAreaElement | any) => {
```
Comment (thedavidprice) [inline packages/forms/src/index.tsx:276]: Using `any` was the only type that didn't give an error. I'm no TS expert so assuming there's a better way to handle this.

Note: multiple occurrences below as well.

**Comment by morganmspencer** [comment]
@thedavidprice it looks like the `register` function with the spread operator is returning the `ref` attribute (as well as `onChange`, `onBlur` and `name`). I think this may have been causing the error with `ref`, so I used the `register` call separately and defined the `onChange` and `onBlur` attributes explicitly.

I could be way off base with this way but 6 out of 12 tests now pass as opposed to the 3 from before. 🤷‍♂️

Would love any help with fixing the other tests.

**Comment by thedavidprice** [comment]
Thanks @morganmspencer Let's see if @cannikin can help get this to the finish line when he's back next Monday. Just ping us again if you don't hear from either of us before then.

**Comment by cannikin** [comment]
Oh mannnn there's a lot of errors here, and these warnings are equally scary:

> Invalid Float. Form field validation not set.  Returning 'undefined' instead of 'undefined'

They also have this new `Controller` thing? A lot of these errors look like they involve the type coercion stuff, which was added after I initially built out the form stuff. It looks like react-hook-form has type coercion natively now: https://react-hook-form.com/api/useform/register (scroll down to the `valueAsNumber`, `valueAsDate` and `setValueAs` attributes).

This is looking like a much bigger deal than just bumping the required version number. How badly do we want to upgrade to 7.x?

**Comment by thedavidprice** [comment]
> This is looking like a much bigger deal than just bumping the required version number. How badly do we want to upgrade to 7.x?

For v1-rc (aiming for Redwood release v0.34), we are attempting to get all dependencies on the current major version. So upgrading to v7 is critical path at this point. I'd say at a minimum it's worth understanding what's going on and potentially required (beyond the current work in this PR) so we can discuss.

**Comment by cannikin** [comment]
> upgrading to v7 is critical path at this point

image

**Comment by thedavidprice** [comment]
Feeling the pain 😭 --> https://github.com/redwoodjs/redwood/pull/2402

**Comment by cannikin** [comment]
Might be time to relax that "latest major version" requirement!

image

**Comment by jtoar** [comment]
@cannikin while we're talking about it, what do you think of this one? https://github.com/redwoodjs/redwood/issues/1434. If we think it's a good idea, we should probably do it all at the same time.

**Comment by jtoar** [comment]
Note to self: these enums need to be changed (see the discussion here https://github.com/redwoodjs/redwood/pull/2641#discussion_r640352247):

https://github.com/redwoodjs/redwood/blob/e6c941aea36433ebb790c75c62a0a0dfc9edb998/packages/forms/src/index.tsx#L30-L52

**Comment by morganmspencer** [comment]
> Note to self: these enums need to be changed (see the discussion here [#2641 (comment)](https://github.com/redwoodjs/redwood/pull/2641#discussion_r640352247)):
> 
> https://github.com/redwoodjs/redwood/blob/e6c941aea36433ebb790c75c62a0a0dfc9edb998/packages/forms/src/index.tsx#L30-L52

I just added a change for this

**Comment by jtoar** [comment]
@morganmspencer awesome thanks!

**Comment by thedavidprice** [comment]
@morganmspencer thanks for your willingness and patience with this PR! Assuming we can get someone to help guide you through the next steps, are you still available and interested in continuing with this?

**Comment by thedavidprice** [comment]
General Note:
I suggest this PR also include the option 2 recommendation from this issue "Deprecate validation in <Form>" #1434

**Comment by morganmspencer** [comment]
@thedavidprice I'm definitely still interested in working on this one with a bit of guidance

**Comment by thedavidprice** [comment]
@morganmspencer thanks! I've asked @dthyresson to take a look and check back here. Keep a 👀!

**Comment by thedavidprice** [comment]
Hi @morganmspencer We are going to close this PR and pick up the work in #3043 

Appreciate all your help (and patience) and welcome you to join in on the new PR if you'd like. No pressure! Just know your help is always welcome.

**Pull request closed without merging**